shopify-client/src/components/Header.vue

```diff
@@ -35,9 +39,17 @@
 <script>
 export default {
   name: "Header",
+  methods:{
+    submitSearch: function(){
+      let keyword = "";
+      if(this.keyword){
+        keyword = "?keyword=" + this.keyword;
+      }
+      window.location.href = '/app/shops/List' + keyword;
```

Comment:
Important to note that I am not using vue's router as it would strictly couple the header to the router

Reply:
Could you elaborate? If this a Vue component why wouldn't we want to use tools from Vue?

Reply:
So with a router, the idea is to import the router. This would allow the usage of router.go().
It is discouraged to use it in components, and should be limited to views itself as attaching it in components would make then heavily coupled.

Reply:
I recommend that you don't use window.location.href because it will trigger a page reload. The point about SPA is that we don't to reload pages. I made a comment down there but I will mention it here again. A better option would be to implement a Search Bar component that takes the keyword as a prop.